src/AzureAppConfigurationImpl.ts

```diff
@@ -168,11 +175,44 @@ export class AzureAppConfigurationImpl extends Map<string, any> implements Azure
      */
     async load() {
         await this.#loadSelectedAndWatchedKeyValues();
-
         // Mark all settings have loaded at startup.
         this.#isInitialLoadCompleted = true;
     }
 
+    /**
+     * Construct hierarchical data object from map.
+     */
+    constructConfigurationObject(options?: ConfigurationObjectConstructionOptions): Record<string, any> {
+        const separator = options?.separator ?? ".";
+
+        // construct hierarchical data object from map
+        const data: Record<string, any> = {};
+        for (const [key, value] of this.#configMap) {
+            const segments = key.split(separator);
+            let current = data;
+            // construct hierarchical data object along the path
+            for (let i = 0; i < segments.length - 1; i++) {
+                const segment = segments[i];
+                // undefined or empty string
+                if (!segment) {
+                    throw new Error(`invalid key: ${key}`);
+                }
+                // create path if not exist
+                if (current[segment] === undefined) {
+                    current[segment] = {};
+                }
+                // The path has been occupied by a non-object value, causing ambiguity.
+                if (typeof current[segment] !== "object") {

```

Comment:
It seems like the order of the keys in configMap is important. If there are two key-values: "app.settings.font.color" = "red" and "app.settings.font" = 1

If the key-value "app.settings.font.color" = "red" is handled first, "app.settings.font" = 1 will override it.

Reply:
Good catch! Before the data assignment, we should ensure that the property is not occupied, i.e. its value should be undefined. 
I'll fix that and add a test case for it.